### ISSUE TITLE
MD007: Incorrect indentation check/fix for unordered list inside ordered list item with MkDocs flavor

### ISSUE THREAD

**Issue by sisp** [rvben/rumdl#491]
### rumdl version

0.1.40

### Installation method

Pre-built binary

### Bug description

For MkDocs-flavored Markdown, the `MD007` rule expects 3 instead of 4 spaces of indentation for an unordered list inside an ordered list item.

### Steps to reproduce

1. Create a file `test.md` with this content:

    ```markdown
    # Test

    1. Lorem ipsum dolor sit amet ...

        - ... consectetur adipiscing elit.
    ```

1. Run `rumdl check test.md`

1. See the linter error.

### Expected behavior

There should be no linter error.

### Actual behavior

There is a linter error:

```console
$ rumdl check test.md
test.md:5:1: [MD007] Expected 3 spaces for indent depth 1, found 4 [*]

Issues: Found 1 issues in 1 file (14ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

### Configuration

```toml
[global]
flavor = "mkdocs"
```

### Operating system

Linux

### Additional context

Indenting the unordered list by 4 spaces is valid MkDocs-flavored Markdown:

```console
$ uvx --from markdown markdown_py test.md
<h1>Test</h1>
<ol>
<li>
<p>Lorem ipsum dolor sit amet ...</p>
<ul>
<li>... consectetur adipiscing elit.</li>
</ul>
</li>
</ol>
```

Auto-fixing `test.md`

```console
$ rumdl check --fix test.md
test.md:5:1: [MD007] Expected 3 spaces for indent depth 1, found 4 [fixed]

Fixed: Fixed 1/1 issues in 1 file (10ms)
```

reformats the file to 3 spaces indentation

```diff
 # Test

 1. Lorem ipsum dolor sit amet ...

-    - ... consectetur adipiscing elit.
+   - ... consectetur adipiscing elit.
```

which is incorrect MkDocs-flavored Markdown:

```console
$ uvx --from markdown markdown_py test.md
<h1>Test</h1>
<ol>
<li>
<p>Lorem ipsum dolor sit amet ...</p>
</li>
<li>
<p>... consectetur adipiscing elit.</p>
</li>
</ol>
```

**Comment by rvben** [comment]
Fixed in [v0.1.41](https://github.com/rvben/rumdl/releases/tag/v0.1.41). MD007 now correctly accepts 4-space indentation for unordered lists nested inside ordered list items with MkDocs flavor. Thanks for reporting @sisp!

**Comment by sisp** [comment]
Thank you very much for the super quick fix, @rvben! 🙇
